typesetters/base.lua

```diff
@@ -1446,10 +1446,13 @@ function typesetter:makeHbox (content)
    self.state.hmodeOnly = true
    SILE.process(content)
 
+   self.state.nodes = bidi.splitNodelistIntoBidiRuns(self.state.nodes, self.frame:writingDirection())
```

Comment:
by the way, to we need to put that back into `self.state.nodes`? There's a lot of back-and-forth stuff of that kind, I know, in the code base, but it would be best to avoid it, and just use locals.